{} uses copy constructor, while = does not

Just a little annoyance I encountered while writing to_json functions for a large number of in house classes.  Ideally I'd like it if the {} operator behaved the same as the = operator.

Given the following code:
```cpp
using json = nlohmann::json;

class A {
private:
	std::string          _val;

	// by including the mutex we've implicitly deleted our copy constructor 
	std::recursive_mutex _mutex;

public:
	A() : _val("Hello World") {};

	friend void to_json(json &j, const A &a) {
		j = {
			{"val", a._val}
		};
	}
};



int main(int argc, char* argv[]) {

	json j = {};
	A a;

        // This line compiles and runs completely fine.
        j["data"]["A"] =  a;

        //this line will fail to compile, due to the implicitly deleted A::A(const A&)
	j["data"] = {{"A", a}};
	
        std::cout << j.dump() << std::endl;
	return 0;
}
```

Xcode:

```
main.cpp:36:24: error: call to implicitly-deleted copy constructor of 'A'
    j["data"] = {{"A", a}};
                       ^
main.cpp:13:26: note: copy constructor of 'A' is implicitly deleted because
      field '_mutex' has an inaccessible copy constructor
    std::recursive_mutex _mutex;
                         ^
./json.hpp:6785:22: note: passing argument to parameter 'args' here
    json_ref(Args... args)
                     ^
1 error generated.
```

--- a/src/json.hpp
+++ b/src/json.hpp
@@ -6789,7 +6789,7 @@ class json_ref
     {}
 
     template <class... Args>
-    json_ref(Args... args)
+    json_ref(Args&&... args)
         : owned_value(std::forward<Args>(args)...),
           value_ref(&owned_value),
           is_rvalue(true)

--- a/test/src/unit-regression.cpp
+++ b/test/src/unit-regression.cpp
@@ -36,6 +36,22 @@ using nlohmann::json;
 #include <list>
 #include <cstdio>
 
+namespace
+{
+  struct nocopy
+  {
+    nocopy() = default;
+    nocopy(const nocopy &) = delete;
+
+    int val = 0;
+
+    friend void to_json(json& j, const nocopy& n)
+    {
+      j = {{"val", n.val}};
+    }
+  };
+}
+
 TEST_CASE("regression tests")
 {
     SECTION("issue #60 - Double quotation mark is not parsed correctly")
@@ -1282,4 +1298,12 @@ TEST_CASE("regression tests")
             }
         }
     */
+
+    SECTION("issue #805 - copy constructor is used with std::initializer_list constructor.")
+    {
+      nocopy n;
+      json j;
+      j = {{"nocopy", n}};
+      CHECK(j["nocopy"]["val"] == 0);
+    }
 }

